### PR TITLE
chore(similarity): Move feature flags to list not used in FE

### PR DESCRIPTION
**Pull request by jangjodi**
Add similarity feature flags to list of flags not used in the frontend
This will prevent them from being serialized in project index endpoint, which is slow for the similarity-embeddings flag

### CODE REVIEW

File: src/sentry/api/serializers/models/project.py [hunk 1]
```diff
@@ -78,6 +78,8 @@
     "first-event-severity-calculation",
     "alert-filters",
     "servicehooks",
+    "similarity-embeddings-delete-by-hash",
+    "projects:similarity-embeddings-backfill",
```
Comment (JoshFerge) [inline src/sentry/api/serializers/models/project.py:82]: i think the projects prefix should be removed, so 
```suggestion
    "similarity-embeddings-backfill",
```

Comment (JoshFerge) [reply src/sentry/api/serializers/models/project.py:82]: but, if we're deleting the flag entirely, do we even need this?

Comment (jangjodi) [reply src/sentry/api/serializers/models/project.py:82]: thanks! this is the backfill flag, which I can cleanup/remove in a separate PR (it's still being used though I think it's always true)

Comment (JoshFerge) [reply src/sentry/api/serializers/models/project.py]: ah i see.

**Review by JoshFerge — APPROVED** [review]

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**